libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/MapboxTripStarter.kt

```diff
@@ -0,0 +1,202 @@
+import com.mapbox.navigation.core.replay.route.ReplayRouteSession
+import com.mapbox.navigation.core.replay.route.ReplayRouteSessionOptions
+import com.mapbox.navigation.core.trip.MapboxTripStarter.Companion.getRegisteredInstance
+import com.mapbox.navigation.utils.internal.logI
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+/**
+ * The [MapboxTripStarter] makes it simpler to switch between a trip session and replay.
+ *
+ * This is not able to observe when location permissions change, so you may need to refresh the
+ * state with [refreshLocationPermissions]. Location permissions are not required for replay.
+ *
+ * There should be one instance of this class at a time. For example, an app Activity and car
+ * Session will need to use the same instance. That will be done automatically if you use
+ * [getRegisteredInstance].
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+class MapboxTripStarter internal constructor() : MapboxNavigationObserver {
+
+    private val tripType = MutableStateFlow<MapboxTripStarterType>(
+        MapboxTripStarterType.MapMatching
+    )
+    private val replayRouteSessionOptions = MutableStateFlow(
+        ReplayRouteSessionOptions.Builder().build()
+    )
+    private val isLocationPermissionGranted = MutableStateFlow(false)
+    private var replayRouteTripSession: ReplayRouteSession? = null
+    private var mapboxNavigation: MapboxNavigation? = null
+
+    private lateinit var coroutineScope: CoroutineScope
+
+    /**
+     * Signals that the [mapboxNavigation] instance is ready for use.
+     * @param mapboxNavigation
+     */
+    override fun onAttached(mapboxNavigation: MapboxNavigation) {
+        this.mapboxNavigation = mapboxNavigation
+        coroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main.immediate)
+
+        // Initialize the options to be aware of the location permissions
+        val context = mapboxNavigation.navigationOptions.applicationContext
+        val granted = PermissionsManager.areLocationPermissionsGranted(context)
+        isLocationPermissionGranted.value = granted
+
+        // Observe changes to state
+        observeStateFlow(mapboxNavigation).launchIn(coroutineScope)
+    }
+
+    /**
+     * Signals that the [mapboxNavigation] instance is being detached.
+     * @param mapboxNavigation
+     */
+    override fun onDetached(mapboxNavigation: MapboxNavigation) {
+        coroutineScope.cancel()
+        onTripDisabled(mapboxNavigation)
+        this.mapboxNavigation = null
+    }
+
+    /**
+     * [MapboxTripStarterType] will not work unless location permissions have
```

Comment:
I think you wanted to specify the type here.

Reply:
Thanks for catching! Updated it to the function call [enableMapMatching]